### PR TITLE
Pole landing description

### PR DESCRIPTION
**Pull request by Charlesl0129**
Added urdf description for the landing pole
Added new landing pad consisted of 4 big tags and 4 small tags for pole landing
Added pole.dae for the target in pole landing
Added segmented pole landing target pngs for printing purposes in the future

### CODE REVIEW

File: rotors_description/launch/xacrodisplay_landing_pole.launch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0"?>
```
Comment (archielee) [inline rotors_description/launch/xacrodisplay_landing_pole.launch:1]: are these just for debugging?

File: rotors_description/urdf/landing_pole.xacro [hunk 2]
```diff
@@ -0,0 +1,167 @@
+<?xml version="1.0"?>
+<robot name="target" xmlns:xacro="http://ros.org/wiki/xacro">
+    <xacro:property name="pole_height" value="1.2192" />
+    <xacro:property name="pole_radius" value="0.05" />
+
+    <!-- landing_cube -->
+    <link name="target/landing_cube">
```
Comment (archielee) [inline rotors_description/urdf/landing_pole.xacro:7]: can you add a little note why this is needed (contact surface, etc.)

File: rotors_description/urdf/landing_pole.xacro [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        <collision>
+            <origin xyz="0 0 0" rpy="0 0 0" />
+            <geometry>
+                <box size="0.04 0.04 0.10"/>
+            </geometry>
+        </collision>
+    </link>
+
+    <!-- connect landing_cube to cone -->
+    <joint name="landing_cube_to_cone" type="fixed">
```
Comment (archielee) [inline rotors_description/urdf/landing_pole.xacro:22]: canonical way to name these things is name the joint after the parent link. so for example `arm_link` -> `arm_joint` -> `wrist_link`. we want to make it explicit whether the area is a link or a joint

File: rotors_description/urdf/pelican.xacro [hunk 4]
```diff
@@ -22,7 +22,7 @@
   <xacro:property name="namespace" value="$(arg namespace)" />
   <xacro:property name="rotor_velocity_slowdown_sim" value="10" />
   <xacro:property name="use_mesh_file" value="true" />
-  <xacro:property name="mesh_file" value="package://rotors_description/meshes/dji_matrice_210_v3_chassis.dae" />
+  <xacro:property name="mesh_file" value="package://rotors_description/meshes/dji_matrice_210_v4.dae" />
```
Comment (archielee) [inline rotors_description/urdf/pelican.xacro:25]: so it should still be `_v1` because its a model number, not a revision number. it'll get confusing when we get our new drone which is a DJI matrice 210 V2

File: rotors_gazebo/launch/spawn_target.launch [hunk 5]
```diff
@@ -4,8 +4,8 @@
   <arg name="namespace" default="target"/>
   <arg name="model" default="$(find rotors_description)/urdf/$(arg namespace).xacro"/>
   <arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
-  <arg name="x" default="0.0"/>
-  <arg name="y" default="-10.0"/>
+  <arg name="x" default="2.0"/>
+  <arg name="y" default="2.0"/>
```
Comment (archielee) [inline rotors_gazebo/launch/spawn_target.launch:8]: Don't change the default here, change the argument when you call `spawn_target.launch` in the main launch file. This really should default to all 0s

File: rotors_gazebo/launch/spawn_mav.launch [hunk 6]
```diff
@@ -5,8 +5,8 @@
   <arg name="namespace" default="$(arg mav_name)"/>
   <arg name="model" default="$(find rotors_description)/urdf/$(arg mav_name)_base.xacro"/>
   <arg name="tf_prefix" default="$(optenv ROS_NAMESPACE)"/>
-  <arg name="x" default="0.0"/>
-  <arg name="y" default="-10.75"/>
+  <arg name="x" default="3.2"/>
```
Comment (archielee) [inline rotors_gazebo/launch/spawn_mav.launch:8]: Don't change the default here, change the argument when you call `spawn_mav.launch` in the main launch file. This really should default to all 0s

File: rotors_gazebo/launch/pole_landing_sim.launch [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  </include>
+
+  <group ns="target">
+    <include file="$(find rotors_gazebo)/launch/spawn_pole.launch">
+      <arg name="enable_logging" value="$(arg enable_logging)" />
+      <arg name="enable_ground_truth" value="$(arg enable_ground_truth)" />
+      <arg name="log_file" value="$(arg log_file)"/>
+    </include>
+    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
+    <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
```
Comment (archielee) [inline rotors_gazebo/launch/pole_landing_sim.launch:33]: I don't think you actually need this joint state publisher, it might actually conflict and publish 0 transforms (similar problem to the gimbal bug that I fixed)

File: rotors_description/launch/xacrodisplay_landing_pole.launch (hunk 1 above)
Comment (Charlesl0129) [reply rotors_description/launch/xacrodisplay_landing_pole.launch:1]: Yes, should I remove it?

File: rotors_description/urdf/landing_pole.xacro (hunk 2 above)
Comment (Charlesl0129) [reply rotors_description/urdf/landing_pole.xacro:7]: Added:

File: rotors_description/urdf/landing_pole.xacro (hunk 3 above)
Comment (Charlesl0129) [reply rotors_description/urdf/landing_pole.xacro:23]: "landing_cube_to_cone" renamed to "target/cone_joint"
"cone_to_pole" renamed to "target/base_link_joint"
"pole_to_tag" renamed to "target/tag_joint"

File: rotors_description/urdf/pelican.xacro (hunk 4 above)
Comment (Charlesl0129) [reply rotors_description/urdf/pelican.xacro:25]: I think Ben made changes to these names. Should I change it back to _v1 and overwrite the original?

File: rotors_gazebo/launch/spawn_mav.launch (hunk 6 above)
Comment (Charlesl0129) [reply rotors_gazebo/launch/spawn_mav.launch:8]: Done, did the same to spawn_pole.launch.

File: rotors_gazebo/launch/pole_landing_sim.launch (hunk 7 above)
Comment (Charlesl0129) [reply rotors_gazebo/launch/pole_landing_sim.launch:36]: Removed it and everything ran fine. We are removing this because all joints are fixed?

File: rotors_description/urdf/pelican.xacro (hunk 4 above)
Comment (archielee) [reply rotors_description/urdf/pelican.xacro:25]: Yes, please do that for that mesh file and all other files that use it

File: rotors_description/launch/xacrodisplay_landing_pole.launch (hunk 1 above)
Comment (archielee) [reply rotors_description/launch/xacrodisplay_landing_pole.launch:1]: Yeah don't think they're needed

File: rotors_gazebo/launch/mav_landing_sim.launch [hunk 8]
```diff
@@ -2,7 +2,7 @@
 
 <launch>
   <arg name="mav_name" default="pelican"/>
-  <arg name="world_name" default="test_city"/>
+  <arg name="world_name" default="basic"/>
```
Comment (archielee) [inline rotors_gazebo/launch/mav_landing_sim.launch:5]: Can you undo this change
